lib/public/components/Pagination/amountSelector.js

```diff
@@ -34,61 +36,84 @@ function isPerPageAmountValid(amount) {
 
 /**
  * Amount input component
- * @param {Function} onblur The method to be triggered when an input is blurred
- * @param {Object} model Providing page model to save custom per page value in it
+ * @param {Function} onCustomChoiceConfirm The method to be triggered when the custom choice is confirmed
+ * @param {PaginationModel} paginationModel the pagination's model
  * @returns {vnode} The view of input
  */
-const perPageAmountInputComponent = (onblur, model) => h('input[type=number][placeholder=Custom]', {
+const perPageAmountInputComponent = (onCustomChoiceConfirm, paginationModel) => h('input[type=number][placeholder=Custom]', {
     style: { marginBottom: '4px' },
-    class: `form-control ${!isPerPageAmountValid(model.customPerPage) && 'border-danger danger'}`,
+    class: `form-control ${!isPerPageAmountValid(paginationModel.customItemsPerPage) && 'border-danger danger'}`,
     title: `from ${PER_PAGE_LIMITS.min} to ${PER_PAGE_LIMITS.max}`,
     max: PER_PAGE_LIMITS.max,
     min: PER_PAGE_LIMITS.min,
     oninput: ({ target }) => {
-        model.setCustomPerPage(target.value);
-        model.notify();
+        const customItemsPerPage = parseInt(target.value, 10);
+        if (!isNaN(customItemsPerPage)) {
+            paginationModel.customItemsPerPage = customItemsPerPage;
+        }
     },
-    onblur: ({ target }) => isPerPageAmountValid(target.value) && onblur(target.value),
+    onblur: ({ target }) => isPerPageAmountValid(target.value) && onCustomChoiceConfirm(target.value),
```

Comment:
I think it would be great to add an event listener for pressing `Enter` as well which would then trigger the choice confirmation and hide the selector. 
What do you think?

Reply:
>     * LHC Fill overview, when selecting Infinite and going to the bottom of the page to request more date will automatically go back to top of the table

Probably a bug I added while keeping the branch up to date, I'll have a look

>     * all pages with pagination, if you start from page 4 and select Infinite, it will not not display pages from 1 to 4 anymore

If I am not mistaken this is a bug that existed before, I tried to keep this PR as simple as possible (it already has quite a lot of changes)

Reply:
> I think it would be great to add an event listener for pressing `Enter` as well which would then trigger the choice confirmation and hide the selector. What do you think?

I agree with you. Should I add it in this PR?

Reply:
The first one is fixed, I'll create the tickets